addons/mail/static/src/core/common/mailbox_icon.js

```diff
@@ -0,0 +1,29 @@
+import { Component } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+
+export class MailboxIcon extends Component {
+    static template = "mail.MailboxIcon";
+    static props = ["mailbox", "size?"];
+
+    setup() {
+        super.setup();
+        this.store = useService("mail.store");
+    }
+
+    get containerClass() {
+        return {
+            "o-ms-1_5 me-2 pt-1":
+                !this.store.discuss.isSidebarCompact && this.props.size !== "large",
+            "fs-3": this.props.size === "large",
+        };
+    }
+
+    get iconClass() {
+        return {
+            "fa-lg": this.store.discuss.isSidebarCompact || this.props.size === "large",
+            "fa-inbox": this.props.mailbox.id === "inbox",
+            "fa-star-o": this.props.mailbox.id === "starred",
+            "fa-history": this.props.mailbox.id === "history",
+        };
+    }
```

Comment:
This could be in template, that way we can just open the XML file and see all we need.